### PR TITLE
CORDA-1049 Minor update to Corda API doc to replace notes relevant to 1.0 to 3.0

### PR DESCRIPTION
**Pull request by GavT**
Some API documentation is related to 1.0 so needs updating for 3.0

### CODE REVIEW

File: docs/source/corda-api.rst [hunk 1]
```diff
@@ -24,20 +24,14 @@ Before reading this page, you should be familiar with the :doc:`key concepts of
 Internal APIs and stability guarantees
 --------------------------------------
 
-.. warning:: For Corda 1.0 we do not currently provide a stable wire protocol or support for database upgrades.
-   Additionally, the JSON format produced by the client-jackson module may change in future.
-   Therefore, you should not expect to be able to migrate persisted data from 1.0 to future versions.
+Corda 3.0 provides a stable wire protocol and support for database upgrades therefore you should expect to be able to migrate persisted data from 3.0 to future versions. However, it will be necessary to recompile applications against future versions of the API until we begin offering ABI stability as well.
```
Comment (fenryka) [inline docs/source/corda-api.rst]: upgrades<full stop>Therefore<comma>

File: docs/source/corda-api.rst [hunk 2]
```diff
@@ -24,20 +24,14 @@ Before reading this page, you should be familiar with the :doc:`key concepts of
...
 
-   Additionally, it may be necessary to recompile applications against future versions of the API until we begin offering
-   ABI stability as well. We plan to do this soon after the release of Corda 1.0.
-
-   Finally, please note that the 1.0 release has not yet been security audited. You should not run it in situations
-   where security is required.
+Additionally, please note Corda 3.0 release has been security audited at a rudimentary level therefore you should not run it in situations where security is required.
```
Comment (gendal) [inline docs/source/corda-api.rst]: I wonder if we can reword this. Are there ANY situations where security is not required?  Is it better to say something like HOW to run it if security is required?

Comment (GavT) [reply docs/source/corda-api.rst]: James Brown is giving me some info here. ETA Mar 6

File: docs/source/corda-api.rst [hunk 3]
```diff
@@ -24,20 +24,20 @@ Before reading this page, you should be familiar with the :doc:`key concepts of
...
 --------------------------------------
 
-.. warning:: For Corda 1.0 we do not currently provide a stable wire protocol or support for database upgrades.
-   Additionally, the JSON format produced by the client-jackson module may change in future.
-   Therefore, you should not expect to be able to migrate persisted data from 1.0 to future versions.
+Corda 3.0 provides a stable wire protocol and support for database upgrades. Therefore, you should expect to be able to migrate persisted data from 3.0 to future versions. However, it will be necessary to recompile applications against future versions of the API until we begin offering ABI stability as well.
 
-   Additionally, it may be necessary to recompile applications against future versions of the API until we begin offering
-   ABI stability as well. We plan to do this soon after the release of Corda 1.0.
+Additionally, please note whilst Corda 3.0 release has been security audited at a rudimentary level, to esnure node security is maintained the following best practices should be followed:
```
Comment (fenryka) [inline docs/source/corda-api.rst]: the or this, "whilst Corda 3.0 release" sounds clumsy,

File: docs/source/corda-api.rst [hunk 4]
```diff
@@ -24,20 +24,20 @@ Before reading this page, you should be familiar with the :doc:`key concepts of
...
 
-   Finally, please note that the 1.0 release has not yet been security audited. You should not run it in situations
-   where security is required.
+* Credentials for RPC users, Database connections and Shell users should be created using a secure password generator, preferably from the Linux command line.
```
Comment (fenryka) [inline docs/source/corda-api.rst]: Oxford comma

Comment (fenryka) [reply docs/source/corda-api.rst]: What about customers not running Linux

File: docs/source/corda-api.rst [hunk 5]
```diff
@@ -24,20 +24,20 @@ Before reading this page, you should be familiar with the :doc:`key concepts of
...
+* Nodes should never be deployed using default or development mode credentials.
+* Corda nodes should use one of the supported database platforms, in preference to the default H2 database which is intended for development purposes only. The supported platforms are SQL Server, Azure Server, Oracle, and Postgres.
```
Comment (fenryka) [inline docs/source/corda-api.rst]: Not for Corda, that's R3 Corda, isn't it?

File: docs/source/corda-api.rst (hunk 3 above)
Comment (gendal) [inline docs/source/corda-api.rst]: ensure

File: docs/source/corda-api.rst [hunk 6]
```diff
@@ -24,20 +24,20 @@ Before reading this page, you should be familiar with the :doc:`key concepts of
 Internal APIs and stability guarantees
 --------------------------------------
 
-.. warning:: For Corda 1.0 we do not currently provide a stable wire protocol or support for database upgrades.
-   Additionally, the JSON format produced by the client-jackson module may change in future.
-   Therefore, you should not expect to be able to migrate persisted data from 1.0 to future versions.
+Corda 3.0 provides a stable wire protocol and support for database upgrades. Therefore, you should expect to be able to migrate persisted data from 3.0 to future versions. However, it will be necessary to recompile applications against future versions of the API until we begin offering ABI stability as well.
```
Comment (gendal) [inline docs/source/corda-api.rst]: I thought we DID now offer ABI stability???

File: docs/source/corda-api.rst (hunk 4 above)
Comment (gendal) [inline docs/source/corda-api.rst]: lowercase D and S

File: docs/source/corda-api.rst [hunk 7]
```diff
@@ -24,20 +24,20 @@ Before reading this page, you should be familiar with the :doc:`key concepts of
...
-   ABI stability as well. We plan to do this soon after the release of Corda 1.0.
+Additionally, please note whilst Corda 3.0 release has been security audited at a rudimentary level, to esnure node security is maintained the following best practices should be followed:
 
-   Finally, please note that the 1.0 release has not yet been security audited. You should not run it in situations
-   where security is required.
+* Credentials for RPC users, Database connections and Shell users should be created using a secure password generator, preferably from the Linux command line.
+* Nodes should never be deployed using default or development mode credentials.
+* Corda nodes should use one of the supported database platforms, in preference to the default H2 database which is intended for development purposes only. The supported platforms are SQL Server, Azure Server, Oracle, and Postgres.
+* Nodes should be operated within a secure network (such as a DMZ) that restricts inbound and outbound traffic to only the required ports. Node operators should aim to allow access to P2P, RPC and Database connections only.
```
Comment (gendal) [inline docs/source/corda-api.rst]: Only p2p port from the internet, right?

File: docs/source/corda-api.rst (hunk 1 above)
Comment (GavT) [reply docs/source/corda-api.rst]: Addressed

File: docs/source/corda-api.rst (hunk 2 above)
Comment (GavT) [reply docs/source/corda-api.rst]: Addressed

File: docs/source/corda-api.rst (hunk 3 above)
Comment (GavT) [reply docs/source/corda-api.rst]: Addressed

Comment (GavT) [reply docs/source/corda-api.rst]: Addressed

File: docs/source/corda-api.rst (hunk 4 above)
Comment (GavT) [reply docs/source/corda-api.rst]: Addressed  by removing specific reference to Linux

Comment (GavT) [reply docs/source/corda-api.rst]: Addressed

File: docs/source/corda-api.rst (hunk 5 above)
Comment (GavT) [reply docs/source/corda-api.rst]: Removed this bullet since it is, as you say, and R3C feature

**Comment by GavT** [comment]
@fenryka please check your comments have been addressed

**Comment by fenryka** [comment]
@GavT you'll need to rebase onto V3 to pass the api check

**Pull request merged**